src/ThemedWrappers.h

```diff
@@ -63,7 +69,7 @@ class ThemedWindowWrapper final :
 
    void UpdatePrefs() override
    {
-      PrefsListener::UpdatePrefs();
+      PrefsListenerBase::UpdatePrefs();
```

Comment:
Maybe you could use composition instead of inheritance:  a member of a type inheriting PrefsListener.  Then you would not need to play with type traits to pass through to the base.  Instead UpdatePrefs is directly invoked directly on this member and on the base class to do different unrelated things that can happen in either sequence.

Reply:
... or maybe, even without composition, all you need is:
```
if constexpr(std::is_base_of_v<PrefsListener, WindowBase>)
   WindowBase::UpdatePrefs();
```